dag_in_context/src/greedy_dag_extractor.rs

```diff
@@ -489,18 +497,17 @@ pub fn extract(
     let egraph_info = EgraphInfo::new(&cost_model, egraph, unextractables);
     let extractor_not_linear = &mut Extractor::new(original_prog, termdag);
 
-    let (mut cost_res, mut res) = extract_with_paths(extractor_not_linear, &egraph_info, None);
-    if extractor_not_linear.check_program_is_linear(&res).is_err() {
-        let effectful_nodes_along_path =
-            extractor_not_linear.find_effectful_nodes_in_program(&res, &egraph_info);
-        extractor_not_linear.costs.clear();
-        (cost_res, res) = extract_with_paths(
-            extractor_not_linear,
-            &egraph_info,
-            Some(&effectful_nodes_along_path),
-        );
-        extractor_not_linear.check_program_is_linear(&res).unwrap();
-    }
+    let (_cost_res, res) = extract_with_paths(extractor_not_linear, &egraph_info, None);
```

Comment:
We don't have to undo this optimization, right?

Reply:
This optimization was hiding bugs- lets not have it for now